docs/internal/GeneralArchitectureGuide.md

```diff
@@ -68,6 +68,50 @@ state must ever be reloaded from persisted state.
 
 ## Deprecations
 
+## Backwards Compatibility
+
+Elasticsearch versions are composed of three pieces of information: the major version, the minor version, and the patch version,
+in that order (major.minor.patch). Patch releases are typically bug fixes; minor releases are improvements; and major releases
+contain new features. As an example, each of 8.0.0, 8.3.0 and 8.3.1 specifies an exact release version. They all have the same
+major version (8) and the last two have the same minor version (8.3). Multiversion compatibility within a cluster, or backwards
+compatibility with older version nodes, is guaranteed across specific versions.
+
+### Transport Layer Backwards Compatibility
+
+Elasticsearch nodes can communicate over the network with all older version nodes in the same major release: all the older minor
```

Comment:
Technically not just older nodes, because newer nodes have the same bwc guarantees:

```suggestion
Elasticsearch nodes can communicate over the network with nodes of any other version in the same major release: all the older minor
```

Reply:
I was getting into the weeds, thanks 👍